tools/2024-06-gitops-feature-spec.md

```diff
@@ -0,0 +1,345 @@
+
+Enterprises that use GitOps and also want to use Radius do not currently have a clear path for how to use both technologies in an integrated and complementary way to enable a “better together” experience.
+
+### Top level goals
+<!-- At the most basic level, what are we trying to accomplish? -->
+- Enable enterprises to get the best of both GitOps and Radius for a better-together experience.
+- Provide a consistent Radius + GitOps integration model for both existing and future GitOps platforms.
+
+### Non-goals (out of scope)
+<!-- What are we explicitly not trying to accomplish? -->
+- We are not trying to replace GitOps tools or platforms.  We are trying to integrate Radius with existing GitOps tools and platforms.
+- We are not trying to replace GitOps practices.  We are trying to enable Radius to work well with GitOps practices.
+
+## Customer profile and challenges
+<!-- Define the primary customer and the key problem / pain point we intend to address for that customer. If there are multiple customers or primary and secondary customers, call them out.   -->
+
+### Customer persona(s)
+<!-- Who is the target customer? Include size/org-structure/decision makers where applicable. -->
+**Infrastructure operators / administrators** - In enterprise applications team that use GitOps, operators are responsible for configuring and managing the git repos for infrastructure and application configuration. They configure GitOps policies as well as the Kubernetes manifests, Helm charts, etc. that are applied to the application and the Kubernetes cluster.
+
+**Application developers** - Responsible for designing, developing and maintaining application code. Developers use GitOps by checking files into their git repositories to configure GitOps settings to auto deploy and monitor their application and dependent infrastructure.
+
+**Site Reliability Engineers** - They leverage GitOps by checking in configuration files into their git repositories that add information for the customization of the workload, i.e. replica pod number, strategic merge solutions. 
+
+### Challenge(s) faced by the customer
+<!-- What challenges do the customer face? Why are they experiencing pain and why do current offerings not meet their need? -->
+Before the introduction of GitOps, enterprise application teams struggled with complex deployment processes; environment/application configuration drift; as well as difficulty following best practices for security and disaster recovery.  GitOps enables teams to define continuous deployment workflows and automation to simplify many of these tasks, which add significant value by increasing efficiency, security, creating better developer experiences, and allowing for faster deployments between code changes.
+
```

Comment:
I think one of the challenges users still face is that GitOps is good for things "in Kubernetes" and can't be applied to things "outside Kubernetes". 

There's a gap in the scenario every time a user needs to combine things that can be controlled with GitOps with things that can't. Radius + Recipes can help here.